### DISCUSSION TITLE
Custom weighted distribution.

### DISCUSSION THREAD

**Discussion by mpnewa** [effolkronium/random#38] [Ideas]
Hi effolkronium,

I am wondering if it would be possible to support something similar to:

`... = Random::get<float>({1.1, 1}, {1.2, 2});`    
`... = Random::get<int> ({1, 3}, {4, 4}, {10, 5});`

Where 1, 2 3, 4, 5 are the distribution weights.
An example of pseudocode that could be used:

```
int sum_of_weight = 0;
for(int i=0; i<num_choices; i++) {
   sum_of_weight += choice_weight[i];
}
int rnd = random(sum_of_weight);
for(int i=0; i<num_choices; i++) {
  if(rnd < choice_weight[i])
    return i;
  rnd -= choice_weight[i];
}
assert(!"should never get here");

https://stackoverflow.com/questions/1761626/weighted-random-numbers
```
The complexity would be in the variadic expansions and lists - I think. That said, my c++ is quite rusty at the moment :( .

**Comment by ilqvya** [comment]
An interesting addition... It's possible. I'll think on implementing this, but not have time right now.  Still you can implement it by your own and create PR for me =)=)

**Comment by ghost** [comment]
i dont know

**Reply by ilqvya** [reply]
?
